docs/v2-recipes/developing-and-testing/testing-the-client.md

```diff
@@ -0,0 +1,156 @@
+
+#### 1. Add a test project
+In the `src` folder, create a create a **.Net Standard** library called **Client.Tests**.
+
+```powershell
+cd src
+dotnet new ClassLib -lang F# -o Client.Tests
+cd ..
+dotnet sln add src/Client.Tests
+```
+
+#### 2. Reference the Client project
+Reference the Client project from the Client.Tests project:
+
+```powershell
+dotnet add Client.Tests reference Client
+```
+
+#### 3. Add the Fable.Mocha package to Test project
+Run the following command:
+
+```powershell
+dotnet add Client.Tests package Fable.Mocha
+```
+
+#### 4. Add a test
+Delete the Library.fs file and replace it with a new file called `Tests.fs` to your test project. Add the following code to it:
+```fsharp
+module Tests
+
+open Fable.Mocha
+open Index
+
+let client = testList "Client" [
+    testCase "Hello received" <| fun _ ->
+        let hello = "Hello Test!"
+        let model, _ = init ()
+        let model, _ = update (GotHello hello) model
+
+        Expect.equal hello model.Hello "The hello should be set"
+]
+
+let all =
+    testList "All"
+        [
+            client
+        ]
+
+[<EntryPoint>]
+let main _ = Mocha.runTests all
+```
+
+#### 5. Add Test web page
+
+Add a file called **index.html** to the root of the test project and add the following content to it:
+```html
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>SAFE Client Tests</title>
+    </head>
+    <body>
+    </body>
+</html>
+```
+
+#### 6. Add test webpack config
+
+- Add a new file root directory called **webpack.tests.config.js**.
+
+- Populate it with the contents of a Fable-compatible webpack config template [such as this](https://github.com/fable-compiler/webpack-config-template/blob/master/webpack.config.js).
+
+#### 7. Update the test webpack config
+
+- Replace the `CONFIG` value in the webpack file you just created with the following:
+```fsharp
+var CONFIG = {
+    indexHtmlTemplate: './src/Client.Tests/index.html',
+    fsharpEntry: './src/Client.Tests/Client.Tests.fsproj',
+    outputDir: './src/Client.Tests',
+    assetsDir: './src/Client.Tests',
+    devServerPort: 8081,
+    devServerProxy: undefined,
+    babel: undefined
+}
+```
+
+- Remove all references and code from the webpack file that refers to MiniCssExtractPlugin or CONFIG.cssEntry.
+
+#### 8. Add launch command
+
+Add the following entry to the `scripts` element in the Client project's `package.json` file:
+
+```json
+"test:live": "webpack-dev-server --config webpack.tests.config.js"
+```
+
+#### 9. Build the Tests
+
+Save all changes and build the Client.Test project.
+
+####9. Launch the test website
+
+Run the command
+
+```powershell
+npm install
+```
+followed by
+```powershell
+npm run test:live
+```
+
+Once the build is complete and the website is running, navigate to `http://localhost:8081/` in a web browser. You should see a test results page that looks like this:
+
+<img src="../../../img/mocha-min-results.png"/>
```

Comment:
It'd be nice to explain to also add the option to run the client tests with node or a headless browser so they can be run in CI. [Fable.Jester](https://shmew.github.io/Fable.Jester) also allows you to tests the React views with a node process without spinning a browser.

I myself, for simplicity, most of the time I just use a node process with a fetch polyfill (Just add `importSideEffects "isomorphic-fetch"` before running the tests) to test the API endpoints the same way the client is calling them.

Reply:
I would submit that as a separate recipe (headless tests) - something that we could (should?) add to the FAKE script in the future, too, for CI runs.